store/src/java/com/zimbra/cs/service/util/VolumeConfigUtil.java

```diff
@@ -101,38 +102,133 @@ public static void validateCreateVolumeRequest(CreateVolumeRequest request, Volu
+                    throw VolumeServiceException.INVALID_REQUEST("Account Port can't be negative number", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                }
+
+                // validate url is empty or not
+                if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getUrl())) {
+                    throw VolumeServiceException.INVALID_REQUEST("Url can't be null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                }
+
+                // validate account is empty or not
+                if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getAccount())) {
+                    throw VolumeServiceException.INVALID_REQUEST("Account can't be null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                }
+
+                // validate namespace is empty or not
+                if (StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalOpenIOInfo().getNamespace())) {
+                    throw VolumeServiceException.INVALID_REQUEST("Name Space can't be null", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+                }
             }
+        }
+        if (Volume.StoreType.EXTERNAL.equals(enumStoreType)) {
+            // validate storage type
+            String storageTypeS3 = volInfoRequest.getVolumeExternalInfo().getStorageType();
+            String storageTypeOpenIO = volInfoRequest.getVolumeExternalOpenIOInfo().getStorageType();
 
-            // validate use in frequent access threshold
-            int useInFrequentAccessThreshold = volInfoRequest.getVolumeExternalInfo().getUseInFrequentAccessThreshold();
-            if (0 > useInFrequentAccessThreshold) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume UseInFrequentAccessThreshold can't be negative number", VolumeServiceException.BAD_VOLUME_USE_IN_FREQUENT_ACCESS_THRESHOLD);
+            if (false == storageTypeS3.equalsIgnoreCase(AdminConstants.A_VOLUME_S3) && false == storageTypeOpenIO.equalsIgnoreCase(AdminConstants.A_VOLUME_OPEN_IO)) {
+                throw VolumeServiceException.INVALID_REQUEST("Volume Storage Type can be only S3 or OpenIO", VolumeServiceException.BAD_VOLUME_STORAGE_TYPE);
             }
 
-            // validate global bucket id
-            String globalS3BucketId = volInfoRequest.getVolumeExternalInfo().getGlobalBucketConfigurationId();
-            if (false == extVolInfoHandler.validateGlobalBucketID(globalS3BucketId)) {
-                throw VolumeServiceException.INVALID_REQUEST("Volume GlobalBucketID provided is incorrect, missing or empty", VolumeServiceException.BAD_VOLUME_GLOBAL_BUCKET_ID);
+            if(AdminConstants.A_VOLUME_S3.equalsIgnoreCase(storageTypeS3)) {
+                // validate use in frequent access
+                if (false != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess() &&
```

Comment:
A boolean condition would either true or false, how is it possible it can't be true and false at the same time?

Reply:
value can be true/false/empty and if anything else should throw exception, so just checking not true Or not false isn't enough add a check for empty as well as below
`if (!StringUtil.isNullOrEmpty(volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess()) && false != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess() && true != volInfoRequest.getVolumeExternalInfo().isUseInFrequentAccess())`

Reply:
Removed the validation, since it is a boolean (not a Boolean), therefore, It can't have any value other than false or true